### DISCUSSION TITLE
The problem of requesting cached data from different ip addresses

### DISCUSSION THREAD

**Discussion by louis5974** [AdguardTeam/AdGuardHome#4429] [Q&A]
![微信截图_20220326085229](https://user-images.githubusercontent.com/53338257/160218663-f4f9236d-080c-45cb-90d5-51f0bbe79ccd.png)
The reason I built Adguard Home is for my personal and family use. I have broadband internet at home. The network ip addresses used by the company are different.

Then a question arises:
When requesting domain name resolution for different IP addresses, "Adguard Home" cache results are not shared.

This is reasonable if you are building a public, public DNS server. But for personal internal use within the same city, such as company and home, the physical distance is very close. "Adguard Home" needs to re-request upstream DNS.

Can you improve the functionality in the future, or add a switch that lets us choose between public DNS type or private DNS type. Under the private DNS option, as long as it is the same domain name resolution request, no matter it is any IP address, it can call the cache to respond.

My Adguard Home Version: v0.107.5
Operating System: Centos 8

@ainar-g

**Comment by louis5974** [comment]
@ainar-g

**Comment by ainar-g** [comment]
That shouldn't really happen.  Have these clients been added to the Persistent clients list?  Also, do they use the EDNS0 Client Subnet option in their request?

**Reply by louis5974** [reply]
Hi! Ainar Garipov. thank you for your reply.

My Adguard Home server, built on a cloud server on the public Internet, is not in a local area network. You may not know the internet situation in China. In the case of not disconnecting the Internet, the operator will automatically change the IP address after 48 hours. In China, only the server has a static ip address, and the home network has a dynamic ip address.

I found that the cache can respond normally when the ip address remains unchanged. After the ip address has changed, no response will be taken from the cache. Therefore, a TTL time setting greater than 2 days is redundant.

I didn't set "Identifying clients" and I didn't fill in "Persistent clients". The IP address of the network is assigned by the operator, and the DHCP server also belongs to the operator.

**Reply by louis5974** [reply]
I did not fill in "Persistent Clients". I can only fill in the CIDR range for dynamic ip addresses.

**Reply by louis5974** [reply]
When "optimistic caching" is enabled, devices with different IP addresses can get cached responses.

**Reply by ainar-g** [reply]
@louis5974, have you tried disabling the *Enable EDNS client subnet* option on the *DNS settings* page?  Does that improve the situation?
